src/engine/sparqlExpressions/SparqlExpressionValueGetters.h

```diff
@@ -118,6 +120,53 @@ struct StringValueGetter {
 
   std::optional<string> operator()(IdOrString s,
                                    const EvaluationContext* ctx) const {
+    return std::visit(
+        [self = this, ctx](auto el) { return self->operator()(el, ctx); },
+        std::move(s));
+  }
+};
+using StringValueGetter = StringValueGetterImpl<true>;
+using StringValueGetterRaw = StringValueGetterImpl<false>;
+
+// Value getters for `isIRI`, `isBlank`, and `isLiteral`.
+template <auto isSomethingFunction, auto prefix>
+struct IsSomethingValueGetter {
+  Id operator()(ValueId id, const EvaluationContext* context) const {
+    return Id::makeFromBool(id.getDatatype() == Datatype::VocabIndex &&
+                            std::invoke(isSomethingFunction,
+                                        context->_qec.getIndex().getVocab(),
+                                        id.getVocabIndex()));
+  }
+  Id operator()(const std::string& s, const EvaluationContext*) const {
+    return Id::makeFromBool(s.starts_with(prefix));
+  }
+  Id operator()(IdOrString s, const EvaluationContext* ctx) const {
+    return std::visit([this, ctx](auto el) { return operator()(el, ctx); },
+                      std::move(s));
+  }
+};
+using IsIriValueGetter =
+    IsSomethingValueGetter<&Index::Vocab::isIri,
+                           ad_utility::ConstexprSmallString<2>{"<"}>;
+using IsBlankNodeValueGetter =
+    IsSomethingValueGetter<&Index::Vocab::isBlankNode,
+                           ad_utility::ConstexprSmallString<3>{"_:"}>;
+using IsLiteralValueGetter =
+    IsSomethingValueGetter<&Index::Vocab::isLiteral,
+                           ad_utility::ConstexprSmallString<2>{"\""}>;
+
+// Value getter for `isNumeric`. Regarding which datatypes count as numeric,
+// see https://www.w3.org/TR/sparql11-query/#operandDataTypes .
+struct IsNumericValueGetter {
```

Comment:
Do we actually already have the `BOUND()` expression?
This can be easily added here (works exactly the same as the `IsNumericValueGetter`, just with different datatypes. I am not sure if we already have an `isValidValueGetter` that does the right thing here (should be in this same file if present).